integration-healthcheck/src/main/java/org/symphonyoss/integration/healthcheck/services/ServiceHealthIndicator.java

```diff
@@ -139,11 +138,22 @@ public Health health() {
     String serviceName = getServiceName();
 
     try {
-      IntegrationBridgeService service = serviceInfoCache.get(serviceName);
-      return Health.status(service.getConnectivity()).withDetail(serviceName, service).build();
-    } catch (UncheckedExecutionException | ExecutionException e) {
+      boolean locked = lock.tryLock(LOCK_PERIOD_SECS, TimeUnit.SECONDS);
+      long nextExecution = lastExecution + TimeUnit.SECONDS.toMillis(SERVICE_CACHE_PERIOD_SECS);
+
+      if (!locked || System.currentTimeMillis() > nextExecution) {
```

Comment:
If not locked, it will end up allowing concurrent updates on the service info cache below.